File: clients/mobile/lib/utils/workflow_enabled_util.dart
Map<String, dynamic>? _map(dynamic value) {
  if (value == null) return null;
  if (value is Map<String, dynamic>) return value;
  if (value is Map) return Map<String, dynamic>.from(value);
  return null;
}

bool isWorkflowEnabled(Map<String, dynamic> config) {
  final modules = _map(config['modules']);
  final workflowModule = (modules?['workflow'] as Map?)?['enabled'];
  if (workflowModule == false) return false;
  final workflow = _map(config['workflow']);
  if (workflow?['enabled'] == false) return false;
  return true;
}

const entityStartWorkflowCodes = <String, String>{
  'PRODUCT': 'STOCK_ADJUSTMENT',
};

String? entityStartWorkflowCode(String entityCode) => entityStartWorkflowCodes[entityCode];

File: clients/web/src/app/shared/utils/workflow-enabled.util.ts
/** True when workflow module and engine are enabled in platform config. */
export function isWorkflowEnabled(config: Record<string, unknown>): boolean {
  const modules = config.modules as Record<string, { enabled?: boolean }> | undefined;
  if (modules?.workflow?.enabled === false) {
    return false;
  }
  const workflow = config.workflow as { enabled?: boolean } | undefined;
  if (workflow?.enabled === false) {
    return false;
  }
  return true;
}

/** Entity code → default workflow start action (reference modules). */
export const ENTITY_START_WORKFLOW: Record<string, string> = {
  PRODUCT: 'STOCK_ADJUSTMENT',
};

export function entityStartWorkflowCode(entityCode: string): string | null {
  return ENTITY_START_WORKFLOW[entityCode] ?? null;
}
